iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/tree/impl/ScalarImpl.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.springconfig.tree.impl;
+
+import java.util.Arrays;
+import java.util.List;
+import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.springconfig.tree.api.Scalar;
+import org.sonar.iac.springconfig.tree.api.SyntaxToken;
+
+public class ScalarImpl extends AbstractSpringConfigImpl implements Scalar {
+  private final SyntaxToken value;
+
+  public ScalarImpl(SyntaxToken value) {
+    this.value = value;
+  }
+
+  @Override
+  public SyntaxToken value() {
+    return value;
+  }
+
+  @Override
+  public List<Tree> children() {
+    return Arrays.asList(value);
```

Comment:
`List.of`?

Reply:
I used `Arrays.asList()` to be able to use a null value and not resulting in an NPE.
I'm not sure we ever run into this problem though.

Reply:
I changed to `List.of`, as we don't expect null values